x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts

```diff
@@ -139,26 +141,32 @@ export function registerTransactionDurationRuleType({
                 ...getDocumentTypeFilterForTransactions(
                   searchAggregatedTransactions
                 ),
-                { term: { [SERVICE_NAME]: ruleParams.serviceName } },
-                {
-                  term: {
-                    [TRANSACTION_TYPE]: ruleParams.transactionType,
-                  },
-                },
+                ...termQuery(SERVICE_NAME, ruleParams.serviceName),
+                ...termQuery(TRANSACTION_TYPE, ruleParams.transactionType),
                 ...environmentQuery(ruleParams.environment),
               ] as QueryDslQueryContainer[],
             },
           },
           aggs: {
-            environments: {
-              terms: {
-                field: SERVICE_ENVIRONMENT,
-                missing: ENVIRONMENT_NOT_DEFINED.value,
+            series: {
+              multi_terms: {
+                terms: [
+                  { field: SERVICE_NAME },
+                  {
+                    field: SERVICE_ENVIRONMENT,
+                    missing: ENVIRONMENT_NOT_DEFINED.value,
+                  },
+                  { field: TRANSACTION_TYPE },
+                ],
+                size: 10000,
```

Comment:
We are already limited to sending max 1000 alerts at a time. Would it make sense to change this to 1000 and sort by latency?

Reply:
makes sense to me, but the other threshold rule types define size: 10000 in the same way, so we'll need to update those as well. Where is the 1000 alert limit defined, can i reference it here?

Reply:
> Where is the 1000 alert limit defined, can i reference it here?

It's defined by alerting so it's might not be worth it to reference. 

Agree, the others should be updated as well if possible